egs/iam/v2/local/chain/tuning/run_cnn_e2eali_1a.sh

```diff
@@ -243,3 +244,21 @@ if [ $stage -le 7 ]; then
     --nj $nj --cmd "$cmd" \
     $dir/graph data/test $dir/decode_test || exit 1;
 fi
+
+if [ $stage -le 8 ] && $decode_val; then
```

Comment:
If it's not a hassle, please decode both `val` and `test` in the same stage using a for loop, e.g. `for testset in test $maybe_val; do  ... done` where maybe_val can be set to "val" or empty. See swbd/s5c/local/chain/run_tdnn.sh.